pkg/client/deploymentlogs.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client"
+	"k8s.io/kubernetes/pkg/conversion/queryparams"
+
+	"github.com/openshift/origin/pkg/deploy/api"
+)
+
+// DeploymentLogsNamespacer has methods to work with DeploymentLogs resources in a namespace
+type DeploymentLogsNamespacer interface {
+	DeploymentLogs(namespace string) DeploymentLogInterface
+}
+
+// DeploymentLogInterface exposes methods on DeploymentLogs resources.
+type DeploymentLogInterface interface {
+	Get(name string, opts api.DeploymentLogOptions) (*kclient.Request, error)
+}
+
+// deploymentLogs implements DeploymentLogsNamespacer interface
+type deploymentLogs struct {
+	r  *Client
+	ns string
+}
+
+// newDeploymentLogs returns a deploymentLogs
+func newDeploymentLogs(c *Client, namespace string) *deploymentLogs {
+	return &deploymentLogs{
+		r:  c,
+		ns: namespace,
+	}
+}
+
+// Get gets the deploymentlogs and return a deploymentLog request
+func (c *deploymentLogs) Get(name string, opts api.DeploymentLogOptions) (*kclient.Request, error) {
+	req := c.r.Get().Namespace(c.ns).Resource("deploymentConfigs").Name(name).SubResource("log")
+
+	versioned, err := kapi.Scheme.ConvertToVersion(&opts, "v1")
+	if err != nil {
+		return &kclient.Request{}, err
```

Comment:
not at all, I just consider it more correct since back at the caller if there is an error we wont touch the request